contracts/security-token/RestrictedTransferAgent.sol

```diff
@@ -24,7 +23,16 @@ contract RestrictedTransferAgent is SecurityTransferAgent {
       return value;
     }
 
-    if (KYC.isWhitelisted(to)) {
+    // The following ifs might seem counterintuitive, but this function will
+    // be executed upon every transfer, so we want to make sure that despite of
+    // Solidity's optimization settings, only relevant external calls are made
+    if (KYC.getFlag(to, 0)) {
+      if (KYC.getFlag(from, 0)) {
+        return value;
+      } else {
+        return 0;
+      }
+    } else if (KYC.getFlag(from, 1)) {
```

Comment:
this mode should only be allowed for admin. This doesn't take care of that.

Reply:
Flag #1 is not specifically for admin, it's for every address we would like to grand "push" privileges. We don't know how many account would need that right in the future.

We are anyway the only authority that can give that privilege